fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/models/FedoraResource.java

```diff
@@ -242,4 +242,11 @@ void replaceProperties(final IdentifierConverter<Resource, FedoraResource> idTra
      * @return the node for this object at the version provided
      */
     Node getNodeVersion(String label);
+
+    /**
+     * This method returns the version label of this frozen resource.
+     * If this resource is not frozen, null is returned.
+     * @return version label
+     */
+    String getVersionLabelOfFrozenResource();
```

Comment:
It's not the name (although that is a little weird) it's that we're passing around notions like "frozen" and "unfrozen" which are themselves JCR-specific. Given that this solves a couple of outstanding tickets, I'm not going to worry about it now, but for the long term, we need to look hard at the `kernel-api` module and get rid of this stuff, replacing it with types and notions that actual reflect our modeling commitments.